packages/data-context/src/sources/VersionsDataSource.ts

```diff
@@ -25,14 +25,20 @@ const NPM_CYPRESS_REGISTRY = 'https://registry.npmjs.org/cypress'
 export class VersionsDataSource {
   private _initialLaunch: boolean
   private _currentTestingType: TestingType | null
-  private _latestVersion: Promise<string>
-  private _npmMetadata: Promise<Record<string, string>>
 
   constructor (private ctx: DataContext) {
     this._initialLaunch = true
     this._currentTestingType = this.ctx.coreData.currentTestingType
-    this._latestVersion = this.getLatestVersion()
-    this._npmMetadata = this.getVersionMetadata()
+    this.#ensureData()
+  }
+
+  #ensureData () {
+    this.ctx.update((d) => {
+      d.versionData.latestVersion ??= this.getLatestVersion().catch((e) => undefined)
+      d.versionData.npmMetadata ??= this.getVersionMetadata().catch((e) => undefined)
```

Comment:
Changed these to be stored in the `coreData` so they are cleared between e2e tests, since we currently reuse the same `ctx` class across all tests.

Would probably be useful to look into creating a new "DataContext" class per-spec and just copying over the server details

Reply:
Just to confirm, you mean per-spec for our internal tests, not per-spec for when the user executes specs? Or both?

Reply:
Correct, per-spec for our internal tests